Core/WebCacheManager.swift

```diff
@@ -389,7 +389,7 @@ extension WKWebsiteDataStore: WebCacheManagerDataStore {
     public func preservedCookies(_ preservedLogins: PreserveLogins) async -> [HTTPCookie] {
         let allCookies = await self.httpCookieStore.allCookies()
         return allCookies.filter {
-            preservedLogins.isAllowed(cookieDomain: $0.domain)
+            "duckduckgo.com" == $0.domain || preservedLogins.isAllowed(cookieDomain: $0.domain)
```

Comment:
There’s a similar check in `legacyDataClearing` function. Maybe you can reuse the `cookieDomainsToPreserve` or have an internal/fileprivate interface for the `WebCacheManager` to do the check against the allowed domains?

BTW, it would be awesome to add a unit test for this as well. The existing one (`testWhenClearedThenDDGCookiesAreRetained`) is only executing `legacyDataClearing`.

Reply:
All this code needs to be cleaned up imo.  Just wanted to get a quick fix in for the issue.